l1-contracts/test/portals/TokenPortal.t.sol

```diff
@@ -154,35 +158,85 @@ contract TokenPortalTest is Test {
     assertEq(portalERC20.balanceOf(address(tokenPortal)), 0, "portal should have no assets");
   }
 
-  function testWithdraw() public {
-    uint256 withdrawAmount = 654;
-    portalERC20.mint(address(tokenPortal), withdrawAmount);
-    address _recipient = address(0xdead);
-    bytes32[] memory entryKeys = new bytes32[](1);
-    entryKeys[0] = outbox.computeEntryKey(
+  function _createWithdrawMessageForOutbox(address _designatedCaller)
+    internal
+    view
+    returns (bytes32)
+  {
+    bytes32 entryKey = outbox.computeEntryKey(
       DataStructures.L2ToL1Msg({
         sender: DataStructures.L2Actor({actor: l2TokenAddress, version: 1}),
         recipient: DataStructures.L1Actor({actor: address(tokenPortal), chainId: block.chainid}),
         content: Hash.sha256ToField(
-          abi.encodeWithSignature("withdraw(uint256,address)", withdrawAmount, _recipient)
+          abi.encodeWithSignature(
+            "withdraw(uint256,address,address)", withdrawAmount, recipient, _designatedCaller
+          )
           )
       })
     );
+    return entryKey;
+  }
+
+  function _addWithdrawMessageInOutbox(address _designatedCaller) internal returns (bytes32) {
+    // send assets to the portal
+    portalERC20.mint(address(tokenPortal), withdrawAmount);
 
+    // Create the message
+    bytes32[] memory entryKeys = new bytes32[](1);
+    entryKeys[0] = _createWithdrawMessageForOutbox(_designatedCaller);
     // Insert messages into the outbox (impersonating the rollup contract)
     vm.prank(address(rollup));
     outbox.sendL1Messages(entryKeys);
+    return entryKeys[0];
+  }
 
-    assertEq(portalERC20.balanceOf(_recipient), 0);
+  function testAnyoneCanCallWithdrawIfNoDesignatedCaller(address _caller) public {
```

Comment:
I think you forgot to impersonate, you are just taking a `_caller` variable and not using it beyond assuming it is not address 0. Remember to prank the calls. Consider 

```solidity
vm.assume(_caller != address(0));
vm.startPrank(_caller);
...
vm.stopPrank();
```